tests/server/api/test_workers.py

```diff
@@ -849,6 +849,47 @@ async def test_worker_status_accounts_for_heartbeat_interval(
         assert workers_response.json()[0]["status"] == "OFFLINE"
 
 
+class TestDeleteWorker:
+    async def test_delete_worker(self, client, work_pool, session, db):
+        work_pool_id = work_pool.id
+        deleted_worker_name = "worker1"
+        for i in range(2):
+            insert_stmt = (await db.insert(db.Worker)).values(
+                name=f"worker{i}",
+                work_pool_id=work_pool_id,
+                last_heartbeat_time=pendulum.now(),
+            )
+            await session.execute(insert_stmt)
+            await session.commit()
+
+        response = await client.delete(
+            f"/work_pools/{work_pool.name}/workers/{deleted_worker_name}"
+        )
+        assert response.status_code == status.HTTP_204_NO_CONTENT
+        remaining_workers = await models.workers.read_workers(
+            session=session,
+            work_pool_id=work_pool_id,
+            db=db,
+        )
+        assert deleted_worker_name not in map(lambda x: x.name, remaining_workers)
+
+    async def test_nonexistent_worker(self, client, work_pool, session, db):
+        worker_name = "worker1"
+        work_pool_id = work_pool.id
+        insert_stmt = (await db.insert(db.Worker)).values(
+            name=worker_name,
+            work_pool_id=work_pool_id,
+            last_heartbeat_time=pendulum.now(),
+        )
+        await session.execute(insert_stmt)
+        await session.commit()
+
+        response = await client.delete(
+            f"/work_pools/${work_pool.name}/workers/does-not-exist"
```

Comment:
Not sure if this intentional but this is checking for a work pool that doesn't exist with `${work_pool.name}`. this is still a good test!

Could another one that a valid work pool but non-existent worker gives 404?

Reply:
I've fixed this by creating a new valid workpool. PTAL in latest commit :D